0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+---
+title: Settlement Engine
+draft: 1
+---
+
+# Settlement Engine
+
+## Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Definitions
+
+**Peers** are two participants that share an accounting relationship and have a medium to communicate with one another.
+
+A **transport** is a communication layer for two peers to send messages between one another.
+
+An **account** is a record of accrued obligations between two peers as they transact with one another, denominated in a particular fungible asset.
+
+An **accounting system** tracks a participant's accounts and the net balance owed between peers.
+
+A **settlement** is a reconciliation of a liability.
+
+A **settlement system** is a medium for exchanging value.
+
+## Overview
+
+This specification codifies a **settlement engine**, a common interface between an accounting system and heterogeneous settlement systems.
+
+Compatible settlement engine implementations enable peers to mutually reconcile one's liabilities through integrating with a shared settlement system.
+
+The purpose of a settlement engine is to perform two primary operations:
+
+1. Execute outgoing settlements to a peer
+2. Acknowledge incoming settlements from a peer
+
+Settlements may differ depending on the settlement engine implementation, such as transferring value on the underlying system or ledger, and/or using a custom protocol through which the peers mutually agree to reconcile a liability.
+
+The core test case for a settlement engine implementation is **the sum of amounts an instance is instructed to settle should eventually equal the sum of amounts the peer's instance acknowledges receipt of.**
+
+Varying implementations, arrangements, and conditions (such as connectivity to the ledger) may delay consistency of instructed and acknowledged settlements by an indeterminate amount of time.
+
+## Example Settlement Arrangements
+
+### Simple value transfers
+
+Suppose two peers, Alice and Bob, are operating the same or compatible settlement engine implementations, which settle using a ledger which both Alice and Bob maintain accounts on.
```

Comment:
I removed all of the examples. I think it became hard to draw the line between what's requisite for implementers, and what might be useful only if you're implementing a particular type of settlement engine. My thought is the latter makes more sense as tutorial content or another medium where it can be more frequently updated and less canonical.

Thoughts on this change? (To better illustrate the flow, I also want to add several diagrams, which may help fill the gap).